sqlalchemy-stubs/orm/query.pyi

```diff
@@ -63,18 +67,17 @@ class Query(object):
     def select_entity_from(self, from_obj): ...
     def __getitem__(self, item): ...
     def slice(self, start, stop): ...
-    def limit(self, limit): ...
+    def limit(self, limit) -> Query[T]: ...
     def offset(self, offset): ...
     def distinct(self, *criterion): ...
     def prefix_with(self, *prefixes): ...
     def suffix_with(self, *suffixes): ...
-    def all(self): ...
+    def all(self) -> List[T]: ...
     def from_statement(self, statement): ...
-    def first(self): ...
+    def first(self) -> T: ...
```

Comment:
Shouldn't this one also be optional?

Reply:
Yes! I still thought `first()` would fail on empty results when I wrote this.